presto-parser/src/main/antlr4/com/facebook/presto/sql/parser/SqlBase.g4

```diff
@@ -223,6 +223,7 @@ externalRoutineName
 queryNoWith:
       queryTerm
       (ORDER BY sortItem (',' sortItem)*)?
+      (FETCH FIRST fetchFirstNRows=INTEGER_VALUE ROWS ONLY)?
       (OFFSET offset=INTEGER_VALUE (ROW | ROWS)?)?
       (LIMIT limit=(INTEGER_VALUE | ALL))?
```

Comment:
we shouldn't allow queries to have both LIMIT and FETCH FIRST

Reply:
This is handled and necessary conditions are in place